### ISSUE TITLE
Doesn't seem to work with ports

### ISSUE THREAD

**Issue by codrinbucur** [UziTech/hyperlink-hyperclick#29]
If the URL looks like this:  "http://site.com:8080/app" two issues happen

1. If the cursor is after "http://site.com" portion of the string and you press CTRL/Command, no hyperlink is detected/created in the UI
2. If the cursor is within the "http://site.com" portion of the string, a link is created for clicking, however only the "http://site.com" portion is copied to the browser and ":8080/app" is left out, which is not what someone would want.

It seems like is unable to parse URL with semicolon/ports ...

Would be great if this could be fixed

Thanks!

**Comment by UziTech** [comment]
seems to work fine for me

![hyperlink](https://user-images.githubusercontent.com/97994/59853006-a0b0c580-9335-11e9-85f0-d44ca1feba4f.gif)

I just wrote a few tests and they are passing as well.

https://github.com/UziTech/hyperlink-hyperclick/blob/ef1c3f8eb58b0abfafe97786e8c9e38b2c8fe04b/spec/test-spec.js#L208-L228

Can you provide the exact code that was making this happen? Maybe there was a weird character in the string, or something with the specific grammar.

**Comment by codrinbucur** [comment]
I am not sure what you mean by the "the exact code". I am just editing an .md or .txt file in Atom, with hyperlinks inside. 

Here is a screenshot of the behaviour in my system:

<img width="284" alt="image" src="https://user-images.githubusercontent.com/10154943/59855803-f4430380-9375-11e9-8a3f-1ee2cab6343a.png">

I have the following config:
Atom 1.38.2 (MacOS)
hyperclick 0.1.5
hyperlink-hyperclick 3.0.0

Let me know if you need more info.

Thanks for looking into it so quickly!

**Comment by codrinbucur** [comment]
I also replaced hyperclick (seems deprecated) with atom-ide-ui, but didn't help

**Comment by UziTech** [comment]
I can't seem to recreate this issue.

![image](https://user-images.githubusercontent.com/97994/59863525-7a495500-934a-11e9-84fd-682e3528b15d.png)

I am using atom-ide-ui on windows but I don't think either of those would affect this package.

**Comment by UziTech** [comment]
I'm trying to go through the code to see if there is anything that might stop it from reading the port as part of the url but the only thing I can think of is that the characters in the url have some strange unicode encoding.

**Comment by UziTech** [comment]
is your file UTF-8 encoded?

**Comment by UziTech** [comment]
@codrinbucur Did you figure out the issue?

**Comment by waffledonkey** [comment]
Firstly thank you for your efforts.  I want to use it more than I do, but I too am experience the same thing -- I think it might be mac specific.  If the URL in the comment is like:

<pre>// https://localhost:8443/search/mivul@neice.sm</pre>

holding down command will either highlight everything prior to the port
<pre>//<strong> https://localhost</strong>:8443/search/mivul@neice.sm</pre>
or if you move the mouse to the other side only a portion of the path
<pre>// https://localhost:8443<strong>/search/mivul</strong>@neice.sm</pre>

and no, it's not specific to the path containing `@` the same thing happens if the path is `/search/635298266`

**Comment by UziTech** [comment]
Unfortunately I don't have access to a mac to test this but the tests do run on macos in [travis-ci](https://travis-ci.org/UziTech/hyperlink-hyperclick/jobs/553084371)

@waffledonkey could this be an issue with encoding or grammar? could you send me a full screen shot of the atom window with the file open?

**Comment by UziTech** [comment]
I'm going to close this since I can't reproduce it. If you have more information feel free to reopen it.

**Comment by dmitrypodshivalov** [comment]
Hi, I have the same issue as the topic-starter on  Windows 10. Many files with md extension have that issue. Only the part of the URL prior colon is highlighted when I hover-over with pressed Ctrl button. 
Atom 1.35.1
Hyperlink-hyperclick 3.0.0
How can I help you to troubleshoot it?

**Comment by UziTech** [comment]
@dmitrypodshivalov 
- Is it just `.md` files that it happens to?
- Can you provide a minimal example?
- Provide a list of all packages you have installed and enabled. (core and third party)
- Put the cursor on the port (the part that is not a link but should be) and execute the `Editor: Log Cursor Scope` command from the command pallette. Provide the scopes that it lists.

**Comment by UziTech** [comment]
@dmitrypodshivalov you should probably also update Atom v1.35.1 was released over 6 months ago.

**Comment by dmitrypodshivalov** [comment]
Hi @UziTech , 
Thanks for the prompt reply and for amazing plugin :)
 
I've updated the Atom to v 1.41.0 x64, unfortuantely it still has the same behavior.
The issue occurs not only in md files, same in txt files and all other extensions.

Here is the example: [atom_hyperclick_test.txt](https://github.com/UziTech/hyperlink-hyperclick/files/3855420/atom_hyperclick_test.txt)
Here is how it looks:
![atom](https://user-images.githubusercontent.com/20282784/69005480-53a44000-0923-11ea-8211-061a8870dad8.gif)

List of packages: [atom_packages.txt](https://github.com/UziTech/hyperlink-hyperclick/files/3855421/atom_packages.txt)

Output of the "Editor: Log Cursor Scope":
```
Scopes at Cursor
text.plain
meta.paragraph.text
markup.underline.link.http.hyperlink

```

**Comment by UziTech** [comment]
can you show me the scopes for the domain part of the link?

**Comment by UziTech** [comment]
Can you also try disabling some of the packages and see if it still happens.

**Comment by dmitrypodshivalov** [comment]
Here it is:
```
Scopes at Cursor

text.plain
meta.paragraph.text
markup.underline.link.http.hyperlink
```

I've disabled all community packages except Hyperlink Hyperclick and atom-ide-ui and restarted atom, the result is the same. in the the URLs with port - only part before colon is highlighted.

**Comment by UziTech** [comment]
Do the following to get more debugging information:
1. uninstall `hyperlink-hyperclick` in Atom
2. Run `git clone -b debug https://github.com/UziTech/hyperlink-hyperclick.git`
3. Run `cd hyperlink-hyperclick`
4. Run `apm link`
5. Open Atom and try a hyperlink with a port.
6. Press <kbd>Ctrl</kbd> + <kbd>Shift</kbd> + <kbd>I</kbd> to open dev tools
7. Share the console log info that starts with `hh`

**Comment by dmitrypodshivalov** [comment]
Hi, 
I've followed the steps you provided, 
and the debug version of the hyperlink-hyperclick works well, it works the same way for urls with and without ports, always correct. 
However Ctrl+Shift+I doesn't start dev console for me, can't find it in the Ctrl+Shift+P menu either.

**Comment by dmitrypodshivalov** [comment]
Here is the log from console:
[-1574027465854.log](https://github.com/UziTech/hyperlink-hyperclick/files/3856151/-1574027465854.log)

**Comment by UziTech** [comment]
Ah, it looks like the bug was fixed in https://github.com/UziTech/hyperlink-hyperclick/commit/ddca1d759756809506fefd6cfbace49694820e9e and never released. I'll release a new version soon.

@dmitrypodshivalov thanks for helping me debug this.

**Comment by UziTech** [comment]
I released v3.0.1 which should fix it

**Comment by dmitrypodshivalov** [comment]
Awesome! Thanks
